invenio_oauthclient/views/client/rest.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Client blueprint used to handle OAuth callbacks."""
+
+from __future__ import absolute_import
+
+from flask import Blueprint, abort, current_app, flash, redirect, request, \
+    url_for
+from flask_oauthlib.client import OAuthException
+from invenio_db import db
+from itsdangerous import BadData, TimedJSONWebSignatureSerializer
+from werkzeug.local import LocalProxy
+
+from invenio_oauthclient._compat import _create_identifier
+from invenio_oauthclient.handlers import set_session_next_url
+from invenio_oauthclient.handlers.rest import response_handler
+from invenio_oauthclient.proxies import current_oauthclient
+from invenio_oauthclient.utils import get_safe_redirect_target
+
+blueprint = Blueprint(
+    'invenio_oauthclient',
+    __name__,
+    url_prefix='/oauth',
+    static_folder='../static',
+    template_folder='../templates',
+)
+
+
+serializer = LocalProxy(
+    lambda: TimedJSONWebSignatureSerializer(
+        current_app.config['SECRET_KEY'],
+        expires_in=current_app.config['OAUTHCLIENT_STATE_EXPIRES'],
+    )
+)
+
+
+@blueprint.record_once
+def post_ext_init(state):
```

Comment:
Probably can be removed, these seem to be specific to the UI settings pages, etc.